src/C++/FieldConvertors.h

```diff
@@ -461,17 +462,29 @@ struct UtcTimeStampConvertor
       }
     }
 
-    return std::string( result, showMilliseconds ? sizeof( result ) : 17 );
+    if( showMicroseconds )
+    {
+      if( integer_to_string_padded ( result + 21, 3, micros )
+          != result + 21 )
+      {
+        throw FieldConvertError();
+      }
+    }
+
+    size_t length = showMicroseconds ? sizeof( result ) : showMicroseconds ? 21 : 17;
```

Comment:
You have a copy/paste mistake here.
size_t length = showMicroseconds ? sizeof( result ) : **showMilliseconds**? 21 : 17;